src/spatialdata/transformations/_utils.py

```diff
@@ -132,16 +115,43 @@ def _(e: DataTree, transformations: MappingToCoordinateSystem_t) -> None:
             for k, v in transformations.items():
                 sequence: BaseTransformation = Sequence([scale_transformation, v])
                 new_transformations[k] = sequence
-            _set_transformations_xarray(xdata, new_transformations)
+            _set_transformations(xdata, new_transformations)
         else:
-            _set_transformations_xarray(xdata, transformations)
+            _set_transformations(xdata, transformations)
             old_shape = new_shape
 
 
 @_set_transformations.register(GeoDataFrame)
 @_set_transformations.register(DaskDataFrame)
 def _(e: Union[GeoDataFrame, GeoDataFrame], transformations: MappingToCoordinateSystem_t) -> None:
-    _set_transformations_to_dict_container(e.attrs, transformations)
+    _set_transformations_to_element(e, transformations)
+    # _set_transformations_to_dict_container(e.attrs, transformations)
+
+
+@_get_transformations.register(DataArray)
+def _(e: DataArray) -> MappingToCoordinateSystem_t | None:
+    return _get_transformations_xarray(e)
+
+
+@_get_transformations.register(DataTree)
+def _(e: DataTree) -> MappingToCoordinateSystem_t | None:
+    from spatialdata.models._utils import TRANSFORM_KEY
+
+    if TRANSFORM_KEY in e.attrs:
+        raise ValueError(
```

Comment:
this is a great check btw, I think it can close this issue: https://github.com/scverse/spatialdata/issues/576